lib/nrf_modem_lib/trace_backends/uart/Kconfig

```diff
@@ -8,18 +8,36 @@
 choice NRF_MODEM_LIB_TRACE_BACKEND
 
 config NRF_MODEM_LIB_TRACE_BACKEND_UART
-	bool "UARTE1"
-	# Modem tracing over UART use the UARTE1 as a dedicated peripheral.
-	# This enables the UARTE1 peripheral and includes nrfx UARTE driver.
-	select NRFX_UARTE1
+	bool "UART"
 
 if NRF_MODEM_LIB_TRACE_BACKEND_UART
 
+choice NRF_MODEM_LIB_TRACE_BACKEND_UART_VERSION
+	prompt "UART trace backend version"
+
+config NRF_MODEM_LIB_TRACE_BACKEND_UART_ASYNC
+	bool "UARTE1 ASYNC"
+	select NRFX_UARTE1
+	select DEPRECATED
+	help
+	  Enable synchronous processing for UARTE1 trace backend.
+
 config NRF_MODEM_LIB_TRACE_BACKEND_UART_SYNC
 	bool "UARTE1 SYNC"
+	select NRFX_UARTE1
+	select DEPRECATED
 	help
 	  Enable synchronous processing for UARTE1 trace backend.
 
+config NRF_MODEM_LIB_TRACE_BACKEND_UART_NEW
```

Comment:
Could the new implementation be called something else than "new"? How about for example `NRF_MODEM_LIB_TRACE_BACKEND_UART_ZEPHYR`, that would be a bit more descriptive? And if the old versions are deprecated, could this one be enabled by default?

Reply:
I can rename it to `_ZEPHYR`.
Regarding what is the default, the idea to keep the existing one as the default for now to avoid breaking changes. So that existing documentation and guides, that does not contain information about Device tree changes etc., will still work, but the user will get a deprecation warning.

Reply:
Does `choice` always have a value? Currently, none of the options is marked as default. If `choice` always has a value, probably `ASYNC` gets implicitly selected because it's the first one. I think it would be good to mark it explicitly as the default, if that's the intended behavior.

Reply:
That is correct, I'll mark it as default.